### PR TITLE
chore: fix permissions of files generated in docker

### PR DESCRIPTION
**Pull request by tzdybal**
## Overview

Currently, docker is executed with default privileges (as root), which makes it hard to deal with files generated by `make proto-gen`. As a workaround, files are copied to a new directory (to get correct access rights), and the original directory is a part of .gitignore.

The solution consists of three improvements:
 - -u flag added to docker command to run as the current user
 - XDG_CHACHE_HOME modified, to mitigate access permissions (it's picked up by buf process inside container)
 - buf output directory is set properly, so there is no need for copying

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### Summary by CodeRabbit

- Chore: Updated the script used for generating protocol buffers. The changes ensure that the script runs in a more secure and efficient manner by using the current user's UID and GID, and by setting the cache directory to a temporary location. This will help in maintaining the cleanliness of the project directory and improve the overall development experience. Please note that the copying of protocol buffer files to the target directory has been removed, so you may need to adjust your workflow accordingly.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/rollkit/rollkit
 
-go 1.20
+go 1.21
```
Comment (nashqueue) [inline go.mod:3]: I go update necessary in this pr ?

Comment (tzdybal) [reply go.mod:3]: No, sorry, I'm just using 1.21 normally. I need to downgrade :/

**Review by nashqueue — APPROVED** [review]
JTMB ACK

**Review by Manav-Aggarwal — APPROVED** [review]
LGTM

**Pull request merged**